custom_components/tahoma/tahoma_device.py

```diff
@@ -107,13 +103,23 @@ def device_state_attributes(self) -> Dict[str, Any]:
     @property
     def device_info(self) -> Dict[str, Any]:
         """Return device registry information for this entity."""
+        # Some devices, such as the Smart Thermostat have several devices in one physical device,
+        # with same device url, terminated by '#' and a number.
+        # We use the base device url as the device identifier.
+        if "#" in self.device_url and not self.device_url.endswith("#1"):
+
+            # Only return the url of the base device, to use main device name and model.
+            return {
+                "identifiers": {(DOMAIN, self.get_base_device_url())},
+            }
+
         manufacturer = self.select_state(CORE_MANUFACTURER_NAME_STATE) or "Somfy"
         model = self.select_state(CORE_MODEL_STATE) or self.device.widget
 
         return {
-            "identifiers": {(DOMAIN, self.unique_id)},
-            "manufacturer": manufacturer,
+            "identifiers": {(DOMAIN, self.device_url)},
```

Comment:
Device with `#1` in the url should also return the base url to be merged with subsequent subdevices.

```suggestion
            "identifiers": {(DOMAIN, self.get_base_device_url())},
```

following, `get_base_device_url()` should be modified as well, see following comment.